### PR TITLE
[SPARK-36468][SQL][DOCS] Update docs about ANSI interval literals

### PR DESCRIPTION
**Pull request by MaxGekk**
### What changes were proposed in this pull request?
In the PR, I propose to update the doc page https://spark.apache.org/docs/latest/sql-ref-literals.html#interval-literal, and describe formats of ANSI interval literals.

Screenshot 2021-08-11 at 10 31 36
Screenshot 2021-08-10 at 20 58 04

### Why are the changes needed?
To improve UX with Spark SQL, and inform users about recently added ANSI interval literals.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Manually checked the generated docs:
```
$ SKIP_API=1 SKIP_RDOC=1 SKIP_PYTHONDOC=1 SKIP_SCALADOC=1 bundle exec jekyll build
```

### CODE REVIEW

**Comment by MaxGekk** [comment]
@cloud-fan @sarutak @AngersZhuuuu @beliefer @Peng-Lei May I ask you review this PR.

File: docs/sql-ref-literals.md [hunk 1]
```diff
@@ -439,32 +439,91 @@ SELECT TIMESTAMP '1997-01' AS col;
...
+|INTERVAL DAY|`[+|-]'[+|-]d'`|`INTERVAL -'100' DAY`|
+|INTERVAL DAY TO HOUR|`[+|-]'[+|-]d h'`|`INTERVAL '-100 10' DAY TO HOUR`|
+|INTERVAL DAY TO MINUTE|`[+|-]'[+|-]d h:m'`|`INTERVAL '100 10:30' DAY TO MINUTE`|
+|INTERVAL DAY TO SECOND|`[+|-]'[+|-]d h:m:s.n'`|`INTERVAL '100 10:30:40.999999' DAY TO SECOND`|
+|INTERVAL HOUR|`[+|-]'[+|-]h'`|`INTERVAL '123' HOUR`|
+|INTERVAL HOUR TO MINUTE|`[+|-]'[+|-]h:m'`|`INTERVAL -'-123:10' HOUR TO MINUTE`|
+|INTERVAL HOUR TO SECOND|`[+|-]'[+|-]h:m:s.n'`|`INTERVAL '123:10:59' HOUR TO SECOND`|
+|INTERVAL MINUTE|`[+|-]'[+|-]m'`|`interval '1000' minute`|
+|INTERVAL MINUTE TO SECOND|`[+|-]'[+|-]m:s.n'`|`INTERVAL '1000:01.001' MINUTE TO SECOND`|
+|INTERVAL SECOND|`[+|-]'[+|-]s.n'`|`INTERVAL '1000.000001' SECOND`|
```
[the remark this answers is not in the document]
Comment (sarutak) [reply docs/sql-ref-literals.md:484]: Even if we don't add this table, is it better to list the valid qualifier patterns for `<start field>`, `<end field>` and `single field>` explained [here](https://github.com/apache/spark/pull/33693/files#diff-eccd46a981d192d82f9c54b911c836fe9de0e7fe535123a1915562270b81cdb8R450)?

File: docs/sql-ref-literals.md [hunk 2]
```diff
@@ -439,32 +439,91 @@ SELECT TIMESTAMP '1997-01' AS col;
...
+|---------|-------|------------|
+|INTERVAL YEAR|`[+|-]'[+|-]y'`|`INTERVAL -'2021' YEAR`|
+|INTERVAL YEAR TO MONTH|`[+|-]'[+|-]y-m'`|`INTERVAL '-2021-07' YEAR TO MONTH`|
+|INTERVAL MONTH|`[+|-]'[+|-]m'`|`interval '10' month`|
+
+Formats of supported day-time interval literals:
+
+|`<interval qualifier>`|Interval string pattern|An instance of the literal|
+|---------|----|-------------------|
+|INTERVAL DAY|`[+|-]'[+|-]d'`|`INTERVAL -'100' DAY`|
```
[the remark this answers is not in the document]
Comment (MaxGekk) [reply docs/sql-ref-literals.md:475]: ah, yes. Let me remove INTERVAL.

File: docs/sql-ref-literals.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (MaxGekk) [reply docs/sql-ref-literals.md:484]: Yep, @sarutak is right. We need to show valid sub-types somewhere, and the patterns. Just a couple examples are not enough from my point of view.

File: docs/sql-ref-literals.md [hunk 3]
```diff
@@ -439,32 +439,91 @@ SELECT TIMESTAMP '1997-01' AS col;
...
 
-* **interval_string_value**
-
-     year-month/day-time interval string.
-
 * **interval_unit**
 
     **Syntax:**
 
       YEAR[S] | MONTH[S] | WEEK[S] | DAY[S] | HOUR[S] | MINUTE[S] | SECOND[S] |
```
[the remark this answers is not in the document]
Comment (MaxGekk) [reply docs/sql-ref-literals.md:523]: It depends on the config, correct?

[the remark this answers is not in the document]
Comment (MaxGekk) [reply docs/sql-ref-literals.md:523]: Added a sentence.

File: docs/sql-ref-literals.md [hunk 4]
```diff
@@ -439,32 +439,93 @@ SELECT TIMESTAMP '1997-01' AS col;
...
+|`<interval qualifier>`|Interval string pattern|An instance of the literal|
+|---------|----|-------------------|
+|DAY|`[+|-]'[+|-]d'`|`INTERVAL -'100' DAY`|
+|DAY TO HOUR|`[+|-]'[+|-]d h'`|`INTERVAL '-100 10' DAY TO HOUR`|
+|DAY TO MINUTE|`[+|-]'[+|-]d h:m'`|`INTERVAL '100 10:30' DAY TO MINUTE`|
+|DAY TO SECOND|`[+|-]'[+|-]d h:m:s.n'`|`INTERVAL '100 10:30:40.999999' DAY TO SECOND`|
+|HOUR|`[+|-]'[+|-]h'`|`INTERVAL '123' HOUR`|
+|HOUR TO MINUTE|`[+|-]'[+|-]h:m'`|`INTERVAL -'-123:10' HOUR TO MINUTE`|
+|HOUR TO SECOND|`[+|-]'[+|-]h:m:s.n'`|`INTERVAL '123:10:59' HOUR TO SECOND`|
+|MINUTE|`[+|-]'[+|-]m'`|`interval '1000' minute`|
```
Comment (Peng-Lei) [inline docs/sql-ref-literals.md:482]: nit：interval -> INTERVAL , minute -> MINTUE. Keep consistent

File: docs/sql-ref-literals.md [hunk 5]
```diff
@@ -439,32 +439,93 @@ SELECT TIMESTAMP '1997-01' AS col;
...
+
+Supported year-month interval literals and theirs formats:
+
+|`<interval qualifier>`|Interval string pattern|An instance of the literal|
+|---------|-------|------------|
+|YEAR|`[+|-]'[+|-]y'`|`INTERVAL -'2021' YEAR`|
+|YEAR TO MONTH|`[+|-]'[+|-]y-m'`|`INTERVAL '-2021-07' YEAR TO MONTH`|
+|MONTH|`[+|-]'[+|-]m'`|`interval '10' month`|
+
+Formats of supported day-time interval literals:
```
Comment (AngersZhuuuu) [inline docs/sql-ref-literals.md:471]: Keep consistence with `Supported year-month interval literals and theirs formats:`

File: docs/sql-ref-literals.md [hunk 6]
```diff
@@ -439,32 +439,93 @@ SELECT TIMESTAMP '1997-01' AS col;
...
+  | <seconds value>
+```
+
+Supported year-month interval literals and theirs formats:
+
+|`<interval qualifier>`|Interval string pattern|An instance of the literal|
+|---------|-------|------------|
+|YEAR|`[+|-]'[+|-]y'`|`INTERVAL -'2021' YEAR`|
+|YEAR TO MONTH|`[+|-]'[+|-]y-m'`|`INTERVAL '-2021-07' YEAR TO MONTH`|
+|MONTH|`[+|-]'[+|-]m'`|`interval '10' month`|
```
Comment (AngersZhuuuu) [inline docs/sql-ref-literals.md:469]: use upper case keep consistence.

File: docs/sql-ref-literals.md (hunk 4 above)
Comment (MaxGekk) [reply docs/sql-ref-literals.md:482]: Don't think that the example is the place where need to be consistence. The purpose of examples is to show as many cases as possible.

File: docs/sql-ref-literals.md (hunk 6 above)
Comment (MaxGekk) [reply docs/sql-ref-literals.md:469]: See https://github.com/apache/spark/pull/33693#discussion_r686493854

File: docs/sql-ref-literals.md (hunk 5 above)
Comment (MaxGekk) [reply docs/sql-ref-literals.md:471]: Light deviation in the structure of sentences makes docs less boring.

File: docs/sql-ref-literals.md [hunk 7]
```diff
@@ -439,32 +439,93 @@ SELECT TIMESTAMP '1997-01' AS col;
...
+<interval string> ::= <quote> [ <sign> ] { <year-month literal> | <day-time literal> } <quote>
+<year-month literal> ::= <years value> [ <minus sign> <months value> ] | <months value>
+<day-time literal> ::= <day-time interval> | <time interval>
+<day-time interval> ::= <days value> [ <space> <hours value> [ <colon> <minutes value> [ <colon> <seconds value> ] ] ]
+<time interval> ::= <hours value> [ <colon> <minutes value> [ <colon> <seconds value> ] ]
+  | <minutes value> [ <colon> <seconds value> ]
+  | <seconds value>
+```
+
+Supported year-month interval literals and theirs formats:
```
[the remark this answers is not in the document]
Comment (MaxGekk) [reply docs/sql-ref-literals.md:463]: The purpose of the table below is to demonstrate the valid combination of fields, actually, too. Case sensitivity is not interval specific, and applicable to other SQL statements. It is not worthy to mention it, example is enough.

[the remark this answers is not in the document]
Comment (MaxGekk) [reply docs/sql-ref-literals.md:463]: I think it makes sense to mention the list of valid field names earlier. Let me do that.

**Review by sarutak — APPROVED** [review]

**Comment by sarutak** [comment]
The latest change LGTM.

**Review by cloud-fan — APPROVED** [review]

**Comment by MaxGekk** [comment]
Merging to master/3.2.
Thank you, @sarutak @cloud-fan @AngersZhuuuu @Peng-Lei for review.

**Pull request closed without merging**